File: src/SkillCatalog/api/SkillCatalog.Api/Models/GitHubSubmissionModels.cs
namespace SkillCatalog.Api.Models;

public sealed record GitHubSessionView(bool Authenticated, long? GitHubUserId, string? Login, DateTimeOffset? ExpiresAt);
public sealed record AuthorizationStartResponse(Uri AuthorizationUrl, Guid TransactionId, DateTimeOffset ExpiresAt);
public sealed record SubmissionFileView(string Path, string Operation, string Sha256, long Size);
public sealed record SubmissionIntentView(Guid Id, string ContributionType, string TargetRepository, string DestinationPath, IReadOnlyList<SubmissionFileView> Files, string PullRequestTitle, DateTimeOffset ExpiresAt);
public sealed record ContributionEvidenceView(string Kind,string Label,string? Status,string? Url);
public sealed record ContributionView(Guid Id, string State, string? PullRequestUrl, int? PullRequestNumber, string? FailureCategory, string? RecoveryMessage, DateTimeOffset UpdatedAt, DateTimeOffset? LastReconciledAt, IReadOnlyList<ContributionEvidenceView>? Evidence=null);
public sealed record ApiFailure(string Category, string Message, string? NextAction, string TraceId);

File: src/SkillCatalog/web/src/api/githubSubmissionModels.ts
export type GitHubSession = { authenticated:boolean; githubUserId?:number; login?:string; expiresAt?:string }
export type SubmissionFile = { path:string; operation:'add'|'change'|'delete'; sha256:string; size:number }
export type SubmissionIntent = { id:string; contributionType:'NewSkill'|'Update'; targetRepository:string; destinationPath:string; files:SubmissionFile[]; pullRequestTitle:string; expiresAt:string }
export type ContributionState = 'Preparing'|'ForkReady'|'BranchReady'|'CommitReady'|'PullRequestOpen'|'ChecksPending'|'AwaitingReview'|'Merged'|'Closed'|'RecoveryRequired'
export type ContributionEvidence = { kind:'pull-request'|'check'|'review'; label:string; status?:string; url?:string }
export type Contribution = { id:string; state:ContributionState; pullRequestUrl?:string; pullRequestNumber?:number; failureCategory?:string; recoveryMessage?:string; updatedAt:string; lastReconciledAt?:string; evidence?:ContributionEvidence[] }
export type ApiFailure = { category:string; message:string; nextAction?:string; traceId:string }

